// videogular2/src/core/core.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var vg_player_1 = require("./vg-player/vg-player");
var vg_media_1 = require("./vg-media/vg-media");
var vg_cue_points_1 = require("./vg-cue-points/vg-cue-points");
var vg_api_1 = require("./services/vg-api");
var vg_fullscreen_api_1 = require("./services/vg-fullscreen-api");
var vg_utils_1 = require("./services/vg-utils");
var vg_controls_hidden_1 = require("./services/vg-controls-hidden");
// components
__export(require("./vg-player/vg-player"));
__export(require("./vg-media/vg-media"));
__export(require("./vg-cue-points/vg-cue-points"));
// services
__export(require("./services/vg-api"));
__export(require("./services/vg-fullscreen-api"));
__export(require("./services/vg-utils"));
__export(require("./services/vg-controls-hidden"));
// types
__export(require("./events/vg-events"));
__export(require("./states/vg-states"));
/**
 * @internal
 */
function coreDirectives() {
    return [
        vg_player_1.VgPlayer, vg_media_1.VgMedia, vg_cue_points_1.VgCuePoints
    ];
}
exports.coreDirectives = coreDirectives;
function coreServices() {
    return [
        vg_api_1.VgAPI, vg_fullscreen_api_1.VgFullscreenAPI, vg_utils_1.VgUtils, vg_controls_hidden_1.VgControlsHidden
    ];
}
exports.coreServices = coreServices;
var VgCoreModule = (function () {
    function VgCoreModule() {
    }
    VgCoreModule.decorators = [
        { type: core_1.NgModule, args: [{
                    declarations: coreDirectives(),
                    exports: coreDirectives(),
                    providers: coreServices()
                },] },
    ];
    /** @nocollapse */
    VgCoreModule.ctorParameters = function () { return []; };
    return VgCoreModule;
}());
exports.VgCoreModule = VgCoreModule;
